packages/playground/src/portal/twin_view.vue

```diff
@@ -25,15 +40,65 @@
             </v-col>
           </v-row>
         </v-list>
+        <v-card-actions v-if="updateRelay" class="justify-end">
+          <v-btn color="primary" class="custom-button" @click="editTwin">Edit</v-btn>
+        </v-card-actions>
       </v-card>
     </v-container>
   </div>
 </template>
 
 <script lang="ts" setup>
+import { GridClient } from "@threefold/grid_client";
+import { generatePublicKey } from "@threefold/rmb_direct_client";
+import { Twins } from "@threefold/tfchain_client";
+import { onMounted, ref } from "vue";
+
 import { useProfileManager } from "../stores";
+import { getGrid } from "../utils/grid";
 
 const profileManager = useProfileManager();
+
+const editingTwin = ref(false);
+const relay = ref(profileManager.profile?.relay || "");
+const updateRelay = ref(false);
+
+onMounted(validateEdit);
+async function validateEdit() {
+  const client = new GridClient({ mnemonic: profileManager.profile!.mnemonic, network: window.env.NETWORK });
+  try {
+    await client._connect();
+    const pk = await generatePublicKey(profileManager.profile!.mnemonic);
+    if (profileManager.profile?.relay !== window.env.RELAY_DOMAIN) {
+      updateRelay.value = true;
+    }
+
+    if (profileManager.profile?.pk !== pk) {
+      UpdateRelay();
+    }
+  } catch (e) {
+    console.log(e);
+  }
+}
+
+function editTwin() {
+  console.log("editing twin");
+  editingTwin.value = true;
+}
+
+async function UpdateRelay() {
+  try {
+    const grid = await getGrid(profileManager.profile!);
+    const twins = new Twins(grid!.tfclient);
+    const newPk = await generatePublicKey(profileManager.profile!.mnemonic);
+    await twins.update({ pk: newPk, relay: relay.value });
```

Comment:
we can use the twin module here on the grid client if we already have it.
`grid.twins.update` and it doesn't need pk, it's already updating it